[HotFix] Verify `username` and `provider id` For Institution Authentication And Fix Share ORCiD Issue [CAS-86][CAS-87]

cslzchen opened the pull request:
### Ticket
https://openscience.atlassian.net/browse/CAS-86
https://openscience.atlassian.net/browse/CAS-87

![oauth-no-orcid](https://user-images.githubusercontent.com/3750414/28092649-386ef52a-6662-11e7-90b7-149c9679a203.png)


### Deployment Note
No server side settings update required.

cslzchen committed:
Add logs and error handling for institution
login if provider id or username is null/empty.

icereval reviewed (changes requested):
Looks good, please add a test though.

cslzchen committed:
Add unit tests for institution login for
notifyRemotePrincipalAuthenticated:
    - not null/empty provider id
    - not null/empty username

cslzchen commented on cas-server-support-osf/src/test/java/io/cos/cas/mock/MockNormalizeRemotePrincipal.java:
```diff
@@ -0,0 +1,38 @@
+    }
+
+    @Override
+    protected JSONObject normalizeRemotePrincipal(final OpenScienceFrameworkCredential credential)
```
The reason that I do `extends` with `@Override` is that `mockito` cannot mock private/protected methods. The `MockNormalizeRemotePrincipal` is a mock class of `OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction` with the method `normalizeRemotePrincipal` overridden.

cslzchen commented on cas-server-support-osf/src/test/java/io/cos/cas/mock/MockNotifyRemotePrincipalAuthenticated.java:
```diff
@@ -0,0 +1,28 @@
+public class MockNotifyRemotePrincipalAuthenticated extends MockOsfRemoteAuthenticateAction {
+
+    /** Constructor. */
+    public MockNotifyRemotePrincipalAuthenticated(final CentralAuthenticationService centralAuthenticationService) {
```
Similar to `MockNormalizeRemotePrincipal`.

cslzchen commented on cas-server-support-osf/src/main/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsAction.java:
```diff
@@ -459,7 +469,7 @@ protected PrincipalAuthenticationResult notifyRemotePrincipalAuthenticated(
      * @throws TransformerException a transformer exception
      */
-    private JSONObject normalizeRemotePrincipal(final OpenScienceFrameworkCredential credential)
+    protected JSONObject normalizeRemotePrincipal(final OpenScienceFrameworkCredential credential)
```
`private` -> `protected` for inheritance.

cslzchen commented on cas-server-support-osf/src/test/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsActionTests.java:
```diff
@@ -39,15 +42,53 @@
+        try {
+            osfRemoteAuthenticate.notifyRemotePrincipalAuthenticated(osfCredential);
+        } catch (final AccountException e) {
+            assertEquals(e.getMessage(), "Username (Email) Required");
```
Compare the exact error message. Similar below.

cslzchen commented on cas-server-support-osf/src/test/java/io/cos/cas/authentication/handler/support/OpenScienceFrameworkPrincipalFromRequestRemoteUserNonInteractiveCredentialsActionTests.java:
```diff
@@ -39,15 +42,53 @@
+        osfCredential.setUsername("");
+        osfCredential.setInstitutionId(AbstractTestUtils.CONST_INSTITUTION_ID);
+        try {
+            osfRemoteAuthenticate.notifyRemotePrincipalAuthenticated(osfCredential);
```
`osfRemoteAuthenticate.doExcute()`fails to work here since the one exception we want to verify is abstracted and handled by the flow.

cslzchen committed:
Disable ORCiD login for OAuth (Share).
